src/Plutarch/Extra/Ord.hs

```diff
@@ -0,0 +1,804 @@
+  where
+    go ::
+        Term s (a :--> PList a :--> a :--> PList a :--> PList a) ->
+        Term s a ->
+        Term s (PList a) ->
+        Term s a ->
+        Term s (PList a) ->
+        Term s (PList a)
+    go self xsHead xsTail ysHead ysTail =
+        pmatch (pcompareBy # cmp # xsHead # ysHead) $ \case
+            PGT -> pcons # ysHead #$ phandleList ysTail (pcons # xsHead # xsTail) $
+                \newYsHead newYsTail -> self # xsHead # xsTail # newYsHead # newYsTail
+            _ -> pcons # xsHead #$ phandleList xsTail (pcons # ysHead # ysTail) $
+                \newXsHead newXsTail -> self # newXsHead # newXsTail # ysHead # ysTail
+
+-- Merges two PLists, throwing out duplicates.
+--
+-- Only call this if you are _sure_ that the PLists are sorted according to the
+-- argument comparator!
+pmergeUnsafeNoDupes ::
+    forall (a :: S -> Type) (s :: S).
+    Term s (PComparator a) ->
+    Term s (PList a) ->
+    Term s (PList a) ->
+    Term s (PList a)
+pmergeUnsafeNoDupes cmp xs ys =
+    phandleList xs (pnubUnsafe # cmp # ys) $ \x' xs' ->
+        phandleList ys (pnubUnsafe # cmp # xs) $ \y' ys' ->
+            (pfix #$ plam $ go) # x' # xs' # y' # ys'
+  where
+    go ::
+        Term s (a :--> PList a :--> a :--> PList a :--> PList a) ->
+        Term s a ->
+        Term s (PList a) ->
+        Term s a ->
+        Term s (PList a) ->
+        Term s (PList a)
+    go self xsHead xsTail ysHead ysTail =
+        pmatch (pcompareBy # cmp # xsHead # ysHead) $ \case
+            PLT -> pcons # xsHead #$ phandleList xsTail (pnubUnsafe # cmp #$ pcons # ysHead # ysTail) $
+                \newXsHead newXsTail -> self # newXsHead # newXsTail # ysHead # ysTail
+            PEQ -> pcons # xsHead #$ phandleList xsTail (pnubUnsafe # cmp # ysTail) $
+                \newXsHead newXsTail -> phandleList ysTail (pnubUnsafe # cmp # xsTail) $
+                    \newYsHead newYsTail -> self # newXsHead # newXsTail # newYsHead # newYsTail
+            PGT -> pcons # ysHead #$ phandleList ysTail (pnubUnsafe # cmp #$ pcons # xsHead # xsTail) $
+                \newYsHead newYsTail -> self # xsHead # xsTail # newYsHead # newYsTail
+
+-- Removes all duplicates from a sorted list.
+--
+-- Only call this if you are _sure_ that the PLists are sorted according to the
+-- argument comparator!
+pnubUnsafe ::
+    forall (a :: S -> Type) (s :: S).
+    Term s (PComparator a :--> PList a :--> PList a)
+pnubUnsafe = phoistAcyclic $ plam $ \cmp -> precList2 (go cmp) pnil
+  where
+    go ::
+        forall (s' :: S).
+        Term s' (PComparator a) ->
+        Term s' a ->
+        Term s' a ->
+        Term s' (PList a) ->
+        Term s' (PList a)
+    go cmp x peek rest =
+        pif
+            (pequateBy # cmp # x # peek)
+            rest
+            (pcons # x # rest)
+
+-- Breaks the argument into sorted chunks; currently, these go up to size 4. To
+-- speed up the chunk sorting, we use sorting networks.
+pmergesort ::
```

Comment:
Name is very misleading, its just the starting phase, and a very specific one (with networks up to size 4). Better leave it open to add different alternative starting phases. Should be named `pmergeStart_2_3_4`.

Reply:
Someone with input sizes of only 64 and 12 doesn't need the smaller networks, for example. On the other hand, someone with input size 40 might want network 5, to get 8 lists that can be merged without leftovers (power of 2). So different starting phases for different use cases make sense.

Reply:
Yeah, fair enough.